src/components/tables/EntityTable.tsx

```diff
@@ -10,51 +12,208 @@ import {
     TablePagination,
     TableRow,
     TableSortLabel,
+    TextField,
+    Toolbar,
     Tooltip,
+    Typography,
 } from '@mui/material';
+import { SupabaseClient } from '@supabase/supabase-js';
+import ExternalLink from 'components/shared/ExternalLink';
 import formatDistanceToNow from 'date-fns/formatDistanceToNow';
-import { MouseEvent, useState } from 'react';
+import { ChangeEvent, MouseEvent, useEffect, useState } from 'react';
 import { FormattedMessage, useIntl } from 'react-intl';
-import {
-    DeploymentStatus,
-    Entity,
-    EntityMetadata,
-} from 'stores/ChangeSetStore';
+import { TABLES } from 'services/supabase';
+import { PostgrestError, useClient } from 'supabase-swr';
+
+enum TableStatuses {
+    LOADING = 'LOADING',
+    DATA_FETCHED = 'DATA_FETCHED',
+    NO_EXISTING_DATA = 'NO_EXISTING_DATA',
+    TECHNICAL_DIFFICULTIES = 'TECHNICAL_DIFFICULTIES',
+    UNMATCHED_FILTER = 'UNMATCHED_FILTER',
+}
+
+type TableStatus =
+    | TableStatuses.LOADING
+    | TableStatuses.DATA_FETCHED
+    | TableStatuses.NO_EXISTING_DATA
+    | TableStatuses.TECHNICAL_DIFFICULTIES
+    | TableStatuses.UNMATCHED_FILTER;
+
+type DeploymentStatus = 'ACTIVE' | 'INACTIVE';
 
 type SortDirection = 'asc' | 'desc';
 
 interface Props {
-    entities: Entity[];
-    updateDeploymentStatus: (
-        key: string,
-        deploymentStatus: DeploymentStatus
-    ) => void;
+    noExistingDataContentIds: {
+        header: string;
+        message: string;
+        docLink: string;
+        docPath: string;
+    };
+}
+
+interface Entity {
+    deploymentStatus: DeploymentStatus;
+    name: string;
+    catalogNamespace: string;
+    connectorType: string;
+    dateCreated: string;
+}
+
+interface TableState {
+    status: TableStatus;
+    error?: PostgrestError;
 }
 
 interface TableColumn {
-    field: keyof EntityMetadata | null;
+    field: keyof Entity | null;
     headerIntlKey: string;
 }
 
-function EntityTable({ entities, updateDeploymentStatus }: Props) {
+interface ConnectorInfo {
+    detail: string;
+    image_name: string;
+}
+
+const CONNECTORS_QUERY = `detail, image_name`;
+
+const LIVE_SPECS_QUERY = `spec_type, catalog_name, updated_at, connector_image_name, id`;
+
+function EntityTable({ noExistingDataContentIds }: Props) {
+    const supabaseClient: SupabaseClient = useClient();
+    const intl = useIntl();
+
+    const [tableState, setTableState] = useState<TableState>({
+        status: TableStatuses.LOADING,
+    });
+
+    const [connectors, setConnectors] = useState<any[] | undefined | null>(
```

Comment:
The three possible types this state segment can assume help the logic determine whether the data is being fetched -- when the data is `null` -- or no existing data was found -- when the data is undefined.

This comment applies to the _publications_ state as well.